sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -66,8 +65,7 @@ private[spark] class HiveExternalCatalog(client: HiveClient) extends ExternalCat
     try {
       body
     } catch {
-      case e: NoSuchItemException =>
-        throw new AnalysisException(e.getMessage)
+      case e: AnalysisException => throw new AnalysisException(e.getMessage)
```

Comment:
I am open to either way. The original code is to catch `NoSuchItemException` and throw `AnalysisException`, which I did not want to change any logic in this patch.